File: javascript/packages/flyql/snippets/recipes/price-quantity.mjs
// Filter records by price greater than 50.00 and quantity at most 10.
//
// FlyQL query: price > 50.00 and quantity <= 10

import { parse } from 'flyql'
import { generateWhere, newColumn } from 'flyql/generators/clickhouse'

const result = parse('price > 50.00 and quantity <= 10')

const columns = {
    price: newColumn({ name: 'price', type: 'Float64' }),
    quantity: newColumn({ name: 'quantity', type: 'UInt32' }),
}

const sql = generateWhere(result.root, columns)
console.log(sql)

File: python/snippets/recipes/price_quantity.py
"""Filter records by price greater than 50.00 and quantity at most 10.

FlyQL query: ``price > 50.00 and quantity <= 10``
"""

from flyql import parse
from flyql.generators.clickhouse.column import Column
from flyql.generators.clickhouse.generator import to_sql_where

result = parse("price > 50.00 and quantity <= 10")

columns = {
    "price": Column("price", "Float64"),
    "quantity": Column("quantity", "UInt32"),
}

sql = to_sql_where(result.root, columns)
print(sql)
